test/integration/record_linkage.py

```diff
@@ -0,0 +1,240 @@
+        "construction_year",
+        list(range(VALID_PLANT_YEARS["lower_bound"], VALID_PLANT_YEARS["upper_bound"])),
+        construction_year_error_prob,
+    )
+    generated_df = _modify_categorical(
+        generated_df,
+        "plant_type",
+        list(PLANT_TYPE_CATEGORIES["categories"]),
+        plant_type_error_prob,
+    )
+
+    # Generate vectors of fuel fractions
+    generated_df[_FUEL_COLS] = _generate_fuel_cols(plant_type, size)
+
+    # Add minor noise to fuel fractions
+    for col in _FUEL_COLS:
+        generated_df[col] = _noisify(generated_df[col])
+
+    return generated_df
+
+
+@pytest.fixture
+def mock_ferc1_plants_df():
+    """Returns a test DataFrame for use in generic record linkage testing."""
+    # Creates test dataframes for a bunch of plants and concatenates them
+    return pd.concat(
+        [
+            _generate_random_test_df("fox lake, mn"),
+            _generate_random_test_df("maalaea", capacity_mean=50.0),
+            _generate_random_test_df("colstrip 1 & 2", capacity_mean=700.0),
+            _generate_random_test_df("wyman 4", capacity_mean=600.0, size=5),
+            _generate_random_test_df("mcintosh", capacity_mean=300.0, size=6),
+            _generate_random_test_df("boulevard", capacity_mean=40.0, size=12),
+            _generate_random_test_df("eagle mountain", capacity_mean=400.0, size=11),
+            _generate_random_test_df("eagle", capacity_mean=150.0, size=14),
+            _generate_random_test_df("permian basin", capacity_mean=340.0),
+            _generate_random_test_df("lake hubbard", capacity_mean=450.0),
+            _generate_random_test_df("north lake", capacity_mean=800.0),
+            _generate_random_test_df("stryker creek", capacity_mean=850.0),
+            _generate_random_test_df("sewell creek", capacity_mean=900.0),
+            _generate_random_test_df("southeast chicago", capacity_mean=400.0, size=3),
+            _generate_random_test_df("mohave", capacity_mean=500.0, size=10),
+            _generate_random_test_df("el segundo", capacity_mean=600.0, size=13),
+            _generate_random_test_df("highgrove", capacity_mean=300.0, size=23),
+            _generate_random_test_df("cool water"),
+            _generate_random_test_df("huntington beach"),
+            _generate_random_test_df("long beach"),
+            _generate_random_test_df("san onofre 2&3"),
+            _generate_random_test_df("allen e. kintigh", capacity_mean=150),
+            _generate_random_test_df("hawthorn 6", capacity_mean=150),
+            _generate_random_test_df("venice c.t.", capacity_mean=500),
+            _generate_random_test_df("keystone *", capacity_mean=1872.0, size=3),
+            _generate_random_test_df("keystone", capacity_mean=50.0, size=21),
+            _generate_random_test_df(
+                "keystone 1&2 (3.70%)", capacity_mean=69.3, size=5
+            ),
+        ]
+    ).reset_index()
+
+
+def test_classify_plants_ferc1(mock_ferc1_plants_df):
```

Comment:
I think this testing seems good enough! One last idea I had from that article on Metamorphic Testing: what if you run the clustering once on the test dataframe, then shuffle the input test dataframe, rerun the clustering, and check that assigned IDs are the same.

Reply:
Never mind, I forgot you're starting with DBSCAN and that seems agnostic to the order of the input dataframe.